add new terms for 1.0.2

denismaier opened the pull request:
Message edited by @bwiernik for documentation purposes.

New terms:

- General terms
  - "advance-online-publication"
  - "album"
  - "audio-recording"
  - "film"
  - "henceforth"
  - "loc-cit"
  - "no-place"
  - "no-publisher"
  - "on"
  - "op-cit"
  - "original-work-published"
  - "personal-communication"
  - "podcast"
  - "podcast-episode"
  - "preprint"
  - "radio-broadcast"
  - "radio-series"
  - "radio-series-episode"
  - "special-issue"
  - "special-section"
  - "television-broadcast"
  - "television-series"
  - "television-series-episode"
  - "video"
  - "working-paper"
  - "bce"
  - "ce"
  - "colon"
  - "comma"
  - "semicolon"

- Item type terms
  - "article"
  - "article-journal"
  - "article-magazine"
  - "article-newspaper"
  - "bill"
  - "book"
  - "broadcast"
  - "chapter"
  - "classic"
  - "collection"
  - "dataset"
  - "document"
  - "entry"
  - "entry-dictionary"
  - "entry-encyclopedia"
  - "event"
  - "graphic"
  - "hearing"
  - "interview"
  - "legal_case"
  - "legislation"
  - "manuscript"
  - "map"
  - "motion_picture"
  - "musical_score"
  - "pamphlet"
  - "paper-conference"
  - "patent"
  - "performance"
  - "periodical"
  - "personal_communication"
  - "post"
  - "post-weblog"
  - "regulation"
  - "report"
  - "review"
  - "review-book"
  - "software"
  - "song"
  - "speech"
  - "standard"
  - "thesis"
  - "treaty"
  - "webpage

- Locator terms
  - "act"
  - "appendix"
  - "article-locator"
  - "canon"
  - "elocation"
  - "equation"
  - "rule"
  - "scene"
  - "table"
  - "timestamp"
  - "title-locator"
  - renamed "sub verbo" to "sub-verbo"

- Role terms
  - "chair"
  - "compiler"
  - "contributor"
  - "curator"
  - "executive-producer"
  - "guest"
  - "host"
  - "narrator"
  - "organizer"
  - "performer"
  - "producer"
  - "script-writer"
  - "series-creator"

bwiernik commented on locales-en-US.xml:
```diff
@@ -159,48 +132,48 @@
     <term name="elocation">			 
-      <single></single>
+      <single></single> <!-- ??? -->
       <multiple></multiple>						 
```
```suggestion
      <single>location</single>
      <multiple>locations</multiple>						 
```

bwiernik commented on locales-en-US.xml:
```diff
@@ -159,48 +132,48 @@
     </term>
     <term name="timestamp">			 
-      <single></single>
+      <single></single> <!-- ??? -->
```
@adam3smith I'm thinking leaving the timestamp terms blank by default? APA style doesn't label timestamps other than by the format (hh:mm:ss). I think that's generally the case with styles?

adam3smith replied:
agree.

bwiernik commented on locales-en-US.xml:
```diff
@@ -301,6 +525,19 @@
     </term>
 
     <!-- VERB ROLE FORMS -->
+    <term name="chair" form="verb"></term> <!-- ??? -->
```
```suggestion
    <term name="chair" form="verb">chaired by</term>
```

bwiernik commented on locales-en-US.xml:
```diff
@@ -275,6 +447,58 @@
     </term>
 
     <!-- SHORT ROLE FORMS -->
+    <term name="chair">
```
@adam3smith @bdarcus I could use a hand here. APA doesn't use short forms for any of these media roles. Do other fields?

adam3smith replied:
I think APA may be the only style that uses chair in citations. I'd just set the short form to "chair" as well (I've seen chr. as an acronym, but I don't think it's particularly common or useful).

bwiernik replied:
@adam3smith Could you help out with compiler/curator/producer/etc. below?

bwiernik commented on locales-en-US.xml:
```diff
@@ -301,6 +525,19 @@
     <!-- VERB ROLE FORMS -->
+    <term name="chair" form="verb"></term> <!-- ??? -->
+    <term name="compiler" form="verb">compiled by</term>
+    <term name="contributor" form="verb">contributions by</term>
```
```suggestion
    <term name="contributor" form="verb">with</term>
```

bwiernik commented on locales-en-US.xml:
```diff
@@ -301,6 +525,19 @@
+    <term name="compiler" form="verb">compiled by</term>
+    <term name="contributor" form="verb">contributions by</term>
+    <term name="curator" form="verb">curated by</term>
+    <term name="executive-producer" form="verb">produced by</term>
```
```suggestion
    <term name="executive-producer" form="verb">executive produced by</term>
```

bwiernik commented on locales-en-US.xml:
```diff
@@ -301,6 +525,19 @@
+    <term name="contributor" form="verb">contributions by</term>
+    <term name="curator" form="verb">curated by</term>
+    <term name="executive-producer" form="verb">produced by</term>
+    <term name="guest" form="verb"></term> <!-- ??? -->
```
```suggestion
    <term name="guest" form="verb">with guest</term>
```

@adam3smith What do you think? Or just `with`?

adam3smith replied:
I think this works, yes

bwiernik commented on locales-en-US.xml:
```diff
@@ -301,6 +525,19 @@
+    <term name="curator" form="verb">curated by</term>
+    <term name="executive-producer" form="verb">produced by</term>
+    <term name="guest" form="verb"></term> <!-- ??? -->
+    <term name="host" form="verb">hosted by</term> <!-- ??? -->
```
```suggestion
    <term name="host" form="verb">hosted by</term>
```

bwiernik commented on locales-en-US.xml:
```diff
@@ -313,6 +550,19 @@
     <!-- SHORT VERB ROLE FORMS -->
+    <term name="chair" form="verb-short"></term> <!-- ??? -->
+    <term name="compiler" form="verb-short">compiled by</term>
+    <term name="contributor" form="verb-short">contributions by</term>
```
@adam3smith @rmzelle @fbennett Can you remind me, if a form is included here, but blank, that is treated as "missing" and falls back to the long form, correct? So many of these could be reduced to blank if the default should be fallback?

```suggestion
    <term name="contributor" form="verb-short">with</term>
```

adam3smith replied:
No, it only falls back to long when the term is missing, not when it's blank. 
A blank term means nothing is being rendered (as you do for the timestamp); that extends to short forms as well (just tested with citeproc-js; not sure if the behavior is in the specs -- I don't think so -- but I think it's correct).

bwiernik replied:
@adam3smith Thanks. In that case, could you help out with the short forms for the verbs here as well? I'm not sure how they would generally be abbreviated.

bwiernik reviewed (changes requested):
@adam3smith Some questions for you

adam3smith commented on locales-en-US.xml:
```diff
@@ -25,6 +25,33 @@
+    <term name="film">film</term>
+    <term name="henceforth">henceforth</term>
+    <term name="loc-cit">loc. cit.</term> <!-- like ibid., the abbreviated form is the regular form  -->
+    <term name="no-place">no place</term>
```
I'd like to see evidence on how universal this is used in Latin. Chicago Manual uses n.p. as does MLA 7 (I think they got rid of this entirely in 8+).

bwiernik replied:
If Chicago and MLA do in English, I'm fine with the baseline being that way.

denismaier replied:
Good, so we keep "no place"

adam3smith commented on locales-en-US.xml:
```diff
@@ -25,6 +25,33 @@
+    <term name="loc-cit">loc. cit.</term> <!-- like ibid., the abbreviated form is the regular form  -->
+    <term name="no-place">no place</term>
+    <term name="no-place" form="short">n. p.</term>
+    <term name="no-publisher">no publisher</term>
```
Ditto here: MLA also uses n.p. here (yes, same acronym). Chicago doesn't know this case.

adam3smith commented on locales-en-US.xml:
```diff
@@ -25,6 +25,33 @@
+    <term name="no-publisher" form="short"></term> <!-- ??? -->
+    <term name="on">on</term>
+    <term name="op-cit">op. cit.</term> <!-- like ibid., the abbreviated form is the regular form  -->
+    <term name="original-work-published"></term> <!-- ??? -->
```
or "originally published"?

bwiernik replied:
APA uses "original work published". Anything specific from Chicago or MLA?

adam3smith replied:
Chicago is all over the place. There's "first published" for reprints (14.114) and "originally published" for translations (14.99). 
MLA has "originally published" (https://style.mla.org/magazine-essay-on-web-site/ )

denismaier replied:
Whatever we choose here should this term include an "as"? As in "originally published **as**"?

bwiernik replied:
No, for example, the APA phrasing is "Original work published 1962". Though, this is a good point that we might need an `as` term.

adam3smith commented:
I think I answered all queries & added some comments on @bwiernik 's suggested edits (haven't looked at the full PR, though)

bwiernik commented on locales-en-US.xml:
```diff
@@ -99,6 +131,50 @@
+      <multiple>tables</multiple>						 
+    </term>
+    <term name="timestamp">			 
+      <single></single> <!-- ??? -->
```
```suggestion
      <single></single> <!-- generally blank -->
```

bwiernik commented on locales-en-US.xml:
```diff
@@ -99,6 +131,50 @@
+      <single>table</single>
+      <multiple>tables</multiple>						 
+    </term>
+    <term name="timestamp">			 
```
```suggestion
    <term name="timestamp"> <!-- generally blank -->
```

bwiernik commented on locales-en-US.xml:
```diff
@@ -169,6 +245,50 @@
+      <single></single>
+      <multiple></multiple>						 
+    </term>
+    <term name="timestamp">			 
```
```suggestion
    <term name="timestamp"> <!-- generally blank -->
```

bwiernik commented on locales-en-US.xml:
```diff
@@ -169,6 +245,50 @@
+      <multiple></multiple>						 
+    </term>
+    <term name="elocation">			 
+      <single></single>
+      <multiple></multiple>						 
```
```suggestion
      <single>loc.</single>
      <multiple>locs.</multiple>
```

bwiernik commented on locales-en-US.xml:
```diff
@@ -169,6 +245,50 @@
+      <multiple></multiple>						 
+    </term>
+    <term name="article-locator">			 
+      <single></single>
+      <multiple></multiple>						 
```
```suggestion
      <single>art.</single>
      <multiple>arts.</multiple>
```

bwiernik commented on locales-en-US.xml:
```diff
@@ -169,6 +245,50 @@
+      <multiple></multiple>						 
+    </term>
+    <term name="equation">			 
+      <single></single>
+      <multiple></multiple>						 
```
```suggestion
      <single>eq.</single>
      <multiple>eqs.</multiple>
```

bwiernik commented on locales-en-US.xml:
```diff
@@ -313,6 +550,19 @@
+    <term name="contributor" form="verb-short">contributions by</term>
+    <term name="curator" form="verb-short">curated by</term>
+    <term name="executive-producer" form="verb-short">produced by</term>
+    <term name="guest" form="verb-short"></term> <!-- ??? -->
```
```suggestion
    <term name="guest" form="verb-short">with guest</term>
```

adam3smith commented on locales-en-US.xml:
```diff
@@ -313,6 +550,19 @@
 
     <!-- SHORT VERB ROLE FORMS -->
+    <term name="chair" form="verb-short"></term> <!-- ??? -->
+    <term name="compiler" form="verb-short">compiled by</term>
```
I don't think we'll find an authoritative source here. I don't think any of these are in MLA or Chicago and thus likely also missing from other works. We could just take best guesses? (e.g. "comp. by" is likely right, and I'd abbreviate with as "w.", etc.) Any other ideas?

bwiernik replied:
What do you think of the below?

denismaier commented on locales-en-US.xml:
```diff
@@ -99,6 +131,50 @@
+      <single>canon</single>
+      <multiple>canons</multiple>						 
+    </term>
+    <term name="elocation">			 
```
@bwiernik you've added a short form for this. Shouldn't we then also add a long form for this?
(OTOH, I'm a bit unsure about this having a term at all. Can't this just stay empty, just as timestamps?)

bwiernik replied:
@adam3smith What do you think? I've mostly seen kindle editions, etc. cited with the term "location" when these are cited (versus page numbers)

denismaier replied:
Ok. In that case I'd say we add "location/locations" as the long form. (We shouldn't be to strict here anyway as styles can always override the defaults. And if something doesn't make sense people can always complain and we make reasonable changes later. But for this, folks need a chance to test use this.)

denismaier commented on locales-en-US.xml:
```diff
@@ -169,6 +245,50 @@
+      <single></single>
+      <multiple></multiple>						 
+    </term>
+    <term name="elocation">			 
```
Can't this just stay empty, just as timestamps?

bwiernik commented on locales-en-US.xml:
```diff
@@ -313,6 +550,19 @@
     <!-- SHORT VERB ROLE FORMS -->
+    <term name="chair" form="verb-short"></term> <!-- ??? -->
+    <term name="compiler" form="verb-short">compiled by</term>
+    <term name="contributor" form="verb-short">with</term>
+    <term name="curator" form="verb-short">curated by</term>
+    <term name="executive-producer" form="verb-short">produced by</term>
+    <term name="guest" form="verb-short">with guest</term>
+    <term name="host" form="verb-short">hosted by</term> <!-- ??? -->
+    <term name="narrator" form="verb-short">narrated by</term>
+    <term name="organizer" form="verb-short">organized by</term>
+    <term name="performer" form="verb-short">performed by</term>
+    <term name="producer" form="verb-short">produced by</term>
+    <term name="script-writer" form="verb-short">written by</term>
+    <term name="series-creator" form="verb-short">created by</term>
```
```suggestion
    <term name="contributor" form="verb-short">w.</term>
    <term name="curator" form="verb-short">cur. by</term>
    <term name="executive-producer" form="verb-short">exec. prod. by</term>
    <term name="guest" form="verb-short">w. guest</term>
    <term name="host" form="verb-short">hosted by</term>
    <term name="narrator" form="verb-short">narr. by</term>
    <term name="organizer" form="verb-short">org. by</term>
    <term name="performer" form="verb-short">perf. by</term>
    <term name="producer" form="verb-short">prod. by</term>
    <term name="script-writer" form="verb-short">writ. by</term>
    <term name="series-creator" form="verb-short">cre. by</term>
```

bwiernik commented on locales-en-US.xml:
```diff
@@ -313,6 +550,19 @@
     <term name="editortranslator" form="verb">edited &amp; translated by</term>
 
     <!-- SHORT VERB ROLE FORMS -->
+    <term name="chair" form="verb-short"></term> <!-- ??? -->
+    <term name="compiler" form="verb-short">compiled by</term>
```
```suggestion
    <term name="chair" form="verb-short">chaired by</term>
    <term name="compiler" form="verb-short">comp. by</term>
```

bwiernik commented on locales-en-US.xml:
```diff
@@ -275,6 +447,58 @@
+    <term name="curator">
+      <single></single>
+      <multiple></multiple>
+    </term>
+    <term name="executive-producer">
+      <single></single>
+      <multiple></multiple>
+    </term>
+    <term name="guest">
+      <single></single>
+      <multiple></multiple>
+    </term>
+    <term name="host">
+      <single></single>
+      <multiple></multiple>
+    </term>
+    <term name="narrator">
+      <single></single>
+      <multiple></multiple>
+    </term>
+    <term name="organizer">
+      <single></single>
+      <multiple></multiple>
+    </term>
+    <term name="performer">
+      <single></single>
+      <multiple></multiple>
+    </term>
+    <term name="producer">
+      <single></single>
+      <multiple></multiple>
+    </term>
+    <term name="script-writer">
+      <single></single>
+      <multiple></multiple>
+    </term>
+    <term name="series-creator">
+      <single></single>
+      <multiple></multiple>
+    </term>
```
```suggestion
      <single>chair</single>
      <multiple>chairs</multiple>
    </term>
    <term name="compiler">
      <single>comp.</single>
      <multiple>comps.</multiple>
    </term>
    <term name="contributor">
      <single>contrib.</single>
      <multiple>contribs.</multiple>
    </term>
    <term name="curator">
      <single>cur.</single>
      <multiple>curs.</multiple>
    </term>
    <term name="executive-producer">
      <single>exec. prod.</single>
      <multiple>exec. prods.</multiple>
    </term>
    <term name="guest">
      <single>guest</single>
      <multiple>guests</multiple>
    </term>
    <term name="host">
      <single>host</single>
      <multiple>hosts</multiple>
    </term>
    <term name="narrator">
      <single>narr.</single>
      <multiple>narrs.</multiple>
    </term>
    <term name="organizer">
      <single>org.</single>
      <multiple>orgs.</multiple>
    </term>
    <term name="performer">
      <single>perf.</single>
      <multiple>perfs.</multiple>
    </term>
    <term name="producer">
      <single>prod.</single>
      <multiple>prods.</multiple>
    </term>
    <term name="script-writer">
      <single>writ.</single>
      <multiple>writs.</multiple>
    </term>
    <term name="series-creator">
      <single>cre.</single>
      <multiple>cres.</multiple>
    </term>
```

bwiernik commented on locales-en-US.xml:
```diff
@@ -249,6 +369,58 @@
+      <single>compiler</single>
+      <multiple>compilers</multiple>
+    </term>
+    <term name="contributor">
```
@adam3smith What do you think? Should `contributor` just have the `with` verb form and no regular form, like with `author`?

denismaier replied:
@bwiernik I'd leave it in for now. (I agree that this will be mostly be used like "Jane Smith, with John Doe. *A very important book* ...", but it won't hurt to have that in here as well. Ok?

bwiernik replied:
sure.

bwiernik commented:
@denismaier Can you add the term stubs for each item type (report, book, broadcast, etc.)?

bwiernik commented on locales-en-US.xml:
```diff
@@ -169,6 +245,50 @@
 
     <!-- SHORT LOCATOR FORMS -->
+    <term name="act">			 
+      <single>a.</single> <!-- according to acronyms.com, but is this really common?-->
+      <multiple></multiple>						 
```
```suggestion
      <single>act</single>
      <multiple>acts</multiple>						 
```

Indigobook and SBL don't have an abbreviation for act, so let's go without unless there is some complaint

denismaier replied:
Ok. Should we just delete the short term form? (Will it fall back to the long form in the absence of a short form?)

denismaier replied:
Or should we short form entries that repeat the long form?

bwiernik commented on locales-en-US.xml:
```diff
@@ -169,6 +245,50 @@
+      <single>art.</single>
+      <multiple>arts.</multiple>
+    </term>
+    <term name="canon">			 
+      <single></single>
+      <multiple></multiple>						 
+    </term>
```
```suggestion
    <term name="canon">			 
      <single>canon</single>
      <multiple>canons</multiple>						 
    </term>
```

denismaier replied:
same here.

bwiernik commented on locales-en-US.xml:
```diff
@@ -169,6 +245,50 @@
+      <multiple>eqs.</multiple>
+    </term>
+    <term name="rule">			 
+      <single></single>
+      <multiple></multiple>						 
```
```suggestion
      <single>r.</single>
      <multiple>rr.</multiple>						 
```

denismaier commented:
> @denismaier Can you add the term stubs for each item type (report, book, broadcast, etc.)?

done.

bwiernik commented on locales-en-US.xml:
```diff
@@ -169,6 +245,50 @@
+      <multiple></multiple>						 
+    </term>
+    <term name="title-locator">			 
+      <single></single>
+      <multiple></multiple>						 
```
```suggestion
      <single>tit.</single>
      <multiple>tits.</multiple>
```

bwiernik commented on locales-en-US.xml:
```diff
@@ -249,6 +369,58 @@
+      <multiple>producers</multiple>
+    </term>
+    <term name="script-writer"> <!-- shouldn't that be scriptwriter?-->
+      <single>scriptwriter</single>
+      <multiple>scriptwriters</multiple>
```
```suggestion
      <single>writer</single>
      <multiple>writers</multiple>
```

> `<!-- shouldn't that be scriptwriter?-->`

It was originally just "writer". The "script" was added during review to reduce confusion. I don't have a problem either way, but the variable name that's incorporated into pandoc, etc. at this point is `script-writer`, so we should probably keep it as is.

@adam3smith

denismaier replied:
Pandoc etc can still be changed easily so that's not that important.

bwiernik commented on locales-en-US.xml:
```diff
@@ -169,6 +337,50 @@
+      <single>sc.</single>
+      <multiple>scs.</multiple>						 
+    </term>
+    <term name="table">			 
+      <single>tab.</single>
+      <multiple>tabs.</multiple>						 
```
```suggestion
    <term name="table">			 
      <single>tbl.</single>
      <multiple>tbls.</multiple>						 
```

denismaier commented on locales-en-US.xml:
```diff
@@ -66,16 +93,113 @@
+    <term name="entry-encyclopedia" form="short"></term>
+    <term name="event">event</term>
+    <term name="event" form="short"></term>
+    <term name="figure">figure</term> <!-- there is an existing figure term right? -->
```
@bwiernik Yes, under locator we already have a `figure` term. In other similar cases we've renamed the locator term to `X-locator`, as in `article` vs. `article-locator` or `title` vs. `title-locator`.

denismaier replied:
@bwiernik What do you think about this? Would renaming the existing figure term to `figure-locator` make sense?

denismaier replied:
@bwiernik What do you think about this? Should we rename it to `figure-locator`?

bwiernik replied:
We need terms for both `figure` and `figure-locator` (and `article` and `article-locator`)

denismaier replied:
Ok, I've renamed the locator term to `figure-locator`. We have both terms for the locator and for the item type now.

denismaier commented:
We should be almost ready, but one general question: How do we deal with short forms being identical to the long forms? Omit the short form? Or keep both?

denismaier commented on locales-en-US.xml:
```diff
@@ -66,16 +93,113 @@
+    <term name="treaty"></term>
+    <term name="treaty" form="short"></term>
+    <term name="webpage"></term>
+    <term name="webpage" form="short"></term>
```
> For a default term for `article`, I'm thinking "preprint" for now?

We also have a dedicated term `preprint`.

denismaier commented:
More questions:
- should we now add an `as` term?
- should there be different `original-work-published` terms for reprints and translations?

bwiernik commented:
I think reprinted/translated would be addressed as part of the related items structure, so I'm inclined to not add those until a better mechanism exists.

denismaier commented:
@bwiernik What do you think is still missing? 

Also, repeating my question from above:

> We should be almost ready, but one general question: How do we deal with short forms being identical to the long forms? Omit the short form? Or keep both?

bwiernik commented:
I think the fall back for short form terms is long form? Can you check that?

denismaier commented:
> I think the fall back for short form terms is long form? Can you check that?

I'm not 100% sure about the fallback logic, but looking at the [verb role forms in the German locale](https://github.com/citation-style-language/locales/blob/ec6d62a9e7c049001a55086a495022520989441c/locales-de-CH.xml#L264) I'd say that this must be the case. (Certain short forms aren't defined.) Also: A test with Chicago Note Bibliography confirms that.

It might neverthless be a good idea to just use the long form even for the short form in those cases as this might help with copying the new terms to other locales.

@adam3smith @rmzelle Thoughts?

adam3smith commented:
I could go either way on this, but I think would softly lean against adding short forms across the board: Since we start with English for anything unknown, that creates a lot more work during translation. But don't feel strongly, so if others disagree or have good arguments to include them, I'm happy to be overruled.

bwiernik commented:
I lean toward not including the short forms. The fallback to long forms is part of the specification, so I think it's okay to make the locales leaner by default (and avoid unintended collisions with English).

denismaier commented:
Ok. Looks like that's the way to go. I'll update the PR.

denismaier commented:
@bwiernik I've updated the PR. (Don't know why the checks fail... Can you see why?)
Anyway, I've corrected some mistakes, removed redundant short forms and refactored the item type terms into two distinct lists, one for long forms, the other for short forms---just as we do with the other terms. Could you please add the abbreviations, and delete those where there's no abbreviated form?

adam3smith commented:
I'm pretty sure the automated test would validate against 1.0.1, which obviously fails

denismaier commented:
> I'm pretty sure the automated test would validate against 1.0.1, which obviously fails

(Just checked: this is what the code says: https://github.com/citation-style-language/locales/blob/d5ee85de8e74d4109509014758b6f496a968ff03/spec/locales_spec.rb#L5)

Ok, so failing tests are not necessarily a bad sign in that case.

bwiernik commented:
Okay, these all validate against the 1.0.2 schema

zepinglee commented:
IMHO the trailing spaces (e.g., [this line](https://github.com/denismaier/locales/blob/0c3396c5a7c1d5d3aa01e43bf0a58d0f9b45c069/locales-en-US.xml#L198)) are supposed to be removed. They may cause extra diff lines when modified.

bdarcus commented:
> * added some terms

FWIW, would probably help to itemize these, so they show up in searches in the future.

I, for example, couldn't find an existing issue or PR that added "figure-locator", so seemed like a mistake.
